### PR TITLE
Unstringlytypify

### LINKED ISSUE #51: `WorkspaceMember` is not entirely correct.
In cargo's output, package id is opaque identifier. The proper way to extract package's name, version and stuff is to look it in the `packages` array by the id and inspect the fields. `WorkspaceMember` instead [splits the id](https://docs.rs/cargo_metadata/0.6.0/src/cargo_metadata/lib.rs.html#233-250). This works at the moment, but this is not guaranteed to work.

### PR DESCRIPTION
**Pull request by matklad**
closes #51

cc https://github.com/rust-lang/cargo/pull/6511

### CODE REVIEW

File: tests/selftest.rs [hunk 1]
```diff
@@ -39,21 +38,18 @@ fn metadata() {
...
-        let test_package_metadata: TestPackageMetadata = serde_json::from_value(value.clone())
-            .unwrap();
-        assert_eq!(test_package_metadata, TestPackageMetadata {
-            some_field: true,
-            other_field: "foo".into(),
-        });
-    }
+    let package_metadata = &metadata.packages[0].metadata.as_object()
+        .expect("package.metadata must be a table. \
+        NOTE: This test currently only works on the beta and nightly channel.");
```
Comment (ehuss) [inline tests/selftest.rs]: This line can probably be removed.

File: src/lib.rs [hunk 2]
```diff
@@ -121,41 +121,71 @@ use std::env;
...
+/// It is possible to inspect the `repr` field, if the need arises, but its
+/// precise format is an implementation detail and is subject to change.
+///
+/// `Metadata` can be indexed by `PackageId`.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[serde(transparent)]
+pub struct PackageId {
+    /// The underlying string representation of id.
+    pub repr: String
+}
```
Comment (oli-obk) [inline src/lib.rs]: This move lost the convenience methods that existed on `WorkspaceMember`

Comment (oli-obk) [reply src/lib.rs]: Oh nevermind. I should have read the issue first

Comment (matklad) [reply src/lib.rs]: Yep, they are now basically `metadata[package_id].url` etc

**Pull request merged**